controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationFile.java

```diff
@@ -339,8 +361,13 @@ private File determineMainFile(final String rawName, final String name) {
         if (mainName == null) {
             // Try the basic case, where name is the name
             final File directoryFile = new File(configurationDir, name);
+            // Translate alias if possible
+            String aliasedName = translateFileAlias(name);
```

Comment:
After looking at the changes implemented, I think we should extract the `translateFileAlias` method outside of the ConfigurationFile to avoid any side effects, for example, at this moment the `bootFileName` field class will get the alias name, instead of the translated file name.

If we extract the `translateFileAlias` method and move it to ServerEnvironment.java, we can leave any domain mode-related classes untouched and have this only applied to standalone files. That would also avoid the need to pass Stability to the `ConfigurationFile` constructors. The Stabilitily.enables(...) invocation could be also invoked in this `translateFileAlias`, which can be a static method so it can be easily tested in a Unit Test.

Reply:
Yes, the `bootFileName` would be the alias, that's the way it behaves with values like "last" or "boot" which are kind of aliases already, except they don't look in the base folder.

And these work in domain too, so I don't see why not to make this available in domain as well.

Reply:
> Yes, the bootFileName would be the alias, that's the way it behaves with values like "last" or "boot" which are kind of aliases already, except they don't look in the base folder.

Thanks for discussing it. Right, but concepts like last / boot / initial and so on are already known concepts, so the code should be already prepared to handle any corner cases using the value of `bootFileName` with these values. However, we are now introducing something new, that will be deactivated on the default stability level. I think minimizing the changes could be a good approach, even more, when we have had undiscovered product issues after modifying this ConfigurationFile. I can see logic related to this `bootFileName` class field and don't control it at all, so being conservative is my first approach to this. I could be wrong.


> And these work in domain too, so I don't see why not to make this available in domain as well.

This is a feature that is not available in domain mode. Also, I'm not sure how useful would be for that, maybe having an alias for host-secondary.xml.

The ConfigurationFile class is used on both, standalone and domain mode. I agree it is the logical class to place details related to aliases for server configuration files, but again, since it has shown in the past to be a bit delicate, this is not a domain mode RFE and it would be in community stability level, I'm suggesting to get it out of this class to minimize the changes.

@bstansberry / @darranl An external opinion about what could be the best approach would be very useful for us. Maybe I am being too conservative with this. I am trying to minimize the changes extracting any related logic outside of the ConfigurationFile, however, this class is the natural place for this, so I could be wrong. What's your feeling about this?

Reply:
@yersan Bear in mind that if I extract this to `ServerEnvironment` I have to make sure not to interfere with the special cases, so I have to duplicate all the checks from `ConfigurationFile`:
 https://github.com/wildfly/wildfly-core/blob/c9d07f4835f0695e9e333a6c32e3019d3193f4a1/controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationFile.java#L346-#L360
to make sure this an alias to be translated. Or I would have to change method to work only with the files that we distribute.

Reply:
@michpetrov I think you won't need to extract any logic. The ConfigurationFile uses the argument `name` as the user's provided name of the configuration file to use. My understanding is we need to translate the alias to the expected name and pass this translated name in this argument. The ConfigurationFile will receive the translated name, so from its point of view, there is no alias in place.

Reply:
@yersan in case of e.g. "boot" I have to pass "boot" as the `name` but my implementation would translate it to "standalone-boot.xml".

Reply:
@michpetrov I would prefer to keep the same meaning of these suffixes, they are suffixes with a special meaning (the same for initla, last, v* and so on), so they should not be treated as possible aliases. You would need to fix the log to take them into account.  If the user uses `boot`, this value should be translated to `boot` and passed as an argument to the ConfigurationFile using the `name` argument.

Reply:
That's what I meant, I have to copy over the logic that checks for these values so that I can skip them (only for `ConfigurationFile` to run the same checks again). And I'll have to expose a few of the constants from `ConfigurationFile`, or change the checks into static methods.

Reply:
@michpetrov Sorry, but I am not following you with your approach.

The logic should take into account the user-provided configuration file name, and if the name is an alias that represents a configuration name we distribute (standalone-full-ha.xml, standalone-microprofile.xml and so on), return that one, otherwise return whatever the has provided.

The ConfiguarionFile will be in charge of managing all the boot, initial, last details as it is doing now. Your logic only would translate aliases in specific defined configuration files. I don't think you need to extract any logic / constants to do that, a simple switch should be able to do all the work.

Reply:
@yersan the basic logic is for the method to take the a string like "custom" and return "standalone-custom.xml", except for a few special strings that translate to the files we provide. But this means a string like "boot" would get turned into "standalone-boot.xml" and I have to make sure that doesn't happen.

Reply:
@michpetrov Ok, I see now. I've been all the time thinking only on the simple approach where we were going to add aliases only for the files we distribute (standalone-full-ha.xml, standalone-microprofile.xml and so on).

You already clarified it in the proposal: https://github.com/wildfly/wildfly-proposals/pull/390#issuecomment-2009841010, but I did not pay the correct attention to that.

If the proposal gets approved as it is, I am fine by adding the changes inside of ConfigurationFile, because you would need to process the boot, last, initial, and so on before doing any possible alias translation. I was just expecting this only for the files we distribute.